app/liquid_tags/codesandbox_tag.rb

```diff
@@ -14,18 +15,39 @@ def render(_context)
   private
 
   def parse_id(input)
-    input_no_space = input.delete(" ")
-    if valid_id?(input_no_space)
-      input_no_space
-    else
-      raise StandardError, "Invalid codesandbox Id"
+    id = input.split(" ").first
+    raise StandardError, "Invalid codesandbox ID" unless valid_id?(id)
 
-    end
+    id
   end
 
   def valid_id?(id)
     id =~ /\A[a-zA-Z0-9\-]{0,60}\Z/
   end
+
+  def parse_options(input)
+    _, *options = input.split(" ")
+
+    # Validation
+    validated_options = options.map { |o| valid_option(o) }.reject { |e| e == nil }
+    raise StandardError, "Invalid Options" unless options.empty? || !validated_options.empty?
+
+    query = options.join("&")
+
+    if query.blank?
+      query
+    else
+      "?#{query}"
+    end
+  end
+
+  def valid_option(option)
+    if option.start_with?("initalpath=", "module=")
```

Comment:
This unfortunately is not strict enough and is susceptible to XSS, for example:
`{% codesandbox ppxnl191zx initialpath=\"onmouseover='alert()' %}`

Ideally we have some sort of regex checking that would check for this. For example, with the following Ruby regex:
```ruby
/\A(initialpath=([a-zA-Z0-9\-\_\/\.])+)|(module=([a-zA-Z0-9\-\_\/\.])+)\Z/
```
will allow `initialpath=`(a string of at least 1 length composed of letters, numbers, dashes, underscores, forward slashes, and periods/dots) or `module=`(the same criteria as above).

I think that regex would cover _most_ file path names, so feel free to use this regex suggestion:
```suggestion
    if (option =~ /\A(initialpath=([a-zA-Z0-9\-\_\/\.])+)|(module=([a-zA-Z0-9\-\_\/\.])+)\Z/)&.zero?
```

Reply:
Thanks for pointing this out, I wasn't 100% sure if my validation was good enough and how things were evaluated in regards to using a iframe.  

I will also is need allow % to be used as well to account for URL Encoding for spaces and slashes.

Reply:
I had to change this a bit more than just adding support for %.  This is what I ended up with:

`if (option =~ /\A(initialpath=([a-zA-Z0-9\-\_\/\.\@\%])+)\Z|\A(module=([a-zA-Z0-9\-\_\/\.\@\%])+)\Z/)&.zero?`